### PR TITLE
Product list paging

### LINKED ISSUE #33: Product list paging
Product list is paged.

Research how to integrate paging with the `Scrollbar` component we are using and apply it to products in `ProductsList`

### LINKED ISSUE #75: Search result formatting
- [ ] add spinner when products are loading
- [ ] add some text 'no products found' when no products found

### PR DESCRIPTION
**Pull request by GitQueenZofia**
closes #33 
closes #75

### CODE REVIEW

File: src/components/ProductListScrollBar/ProductListScrollBar.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import Scrollbars, { positionValues } from 'react-custom-scrollbars-2';
+
+type ScrollBarProps = {
+  className?: string;
+  children?: ReactNode;
+  onLoadMore?: () => void;
+  onScroll?: () => void;
+};
+
+const ProductListScrollBar = ({
```
Comment (andrmdmd) [inline src/components/ProductListScrollBar/ProductListScrollBar.tsx:11]: I think the name should be more generic, maybe `PagingScrollBar`?

File: src/components/ProductListScrollBar/ProductListScrollBar.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import React, { ReactNode, useCallback } from 'react';
```
Comment (andrmdmd) [inline src/components/ProductListScrollBar/ProductListScrollBar.tsx:1]: linting

File: package.json [hunk 3]
```diff
@@ -25,6 +25,8 @@
     "react-redux": "^9.1.0",
     "react-router-dom": "^6.22.3",
     "react-scripts": "5.0.1",
+    "react-spinner": "^0.2.7",
+    "react-spinners": "^0.13.8",
```
Comment (andrmdmd) [inline package.json:29]: are we using both?

File: src/components/ProductListScrollBar/ProductListScrollBar.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  className = '',
+  onLoadMore,
+}: ScrollBarProps): JSX.Element => {
+  const handleScroll = (values: positionValues) => {
+    const { scrollTop, scrollHeight, clientHeight } = values;
+
+    const distanceFromBottom = scrollHeight - (scrollTop + clientHeight);
+    console.log(distanceFromBottom);
+
+    if (distanceFromBottom < 50) {
```
Comment (andrmdmd) [inline src/components/ProductListScrollBar/ProductListScrollBar.tsx:22]: I tested it with a few values and i thing something like `300` here has a better feel

**Review by andrmdmd — CHANGES_REQUESTED** [review]

File: src/components/ProductListScrollBar/ProductListScrollBar.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+const ProductListScrollBar = ({
+  children,
+  className = '',
+  onLoadMore,
+}: ScrollBarProps): JSX.Element => {
+  const handleScroll = (values: positionValues) => {
+    const { scrollTop, scrollHeight, clientHeight } = values;
+
+    const distanceFromBottom = scrollHeight - (scrollTop + clientHeight);
+    console.log(distanceFromBottom);
```
Comment (andrmdmd) [inline src/components/ProductListScrollBar/ProductListScrollBar.tsx:20]: I think it is a leftover debugging log

**Review by andrmdmd — APPROVED** [review]

**Comment by andrmdmd** [comment]
GJ!

**Review by mslup — APPROVED** [review]

**Pull request merged**